packages/core/src/components/tag/_common.scss

```diff
@@ -59,84 +86,49 @@ $tag-padding-large: ($tag-height-large - $pt-icon-size-large) / 2 !default;
     border-radius: $tag-height-large / 2;
   }
 
-  #{$removable-selector} {
+  #{$tag-removable-selector} {
     padding-right: $tag-height-large;
+
+    .pt-tag-remove {
+      @include pt-icon-sized($pt-icon-size-large);
+
+      padding: $tag-padding-large;
+    }
   }
 }
 
 @mixin tag-minimal() {
   &:not([class*="pt-intent-"]) {
-    background: rgba($gray3, 0.2);
+    background-color: rgba($gray3, 0.2);
     color: $pt-text-color;
 
     .pt-dark & {
-      background: rgba($gray3, 0.2);
       color: $pt-dark-text-color;
     }
   }
 
   @each $intent, $color in $pt-intent-colors {
     &.pt-intent-#{$intent} {
-      background: rgba($color, 0.15);
+      background-color: rgba($color, 0.2);
       color: map-get($pt-intent-text-colors, $intent);
-    }
 
-    .pt-dark &.pt-intent-#{$intent} {
-      background: rgba($color, 0.2);
-      color: map-get($pt-dark-intent-text-colors, $intent);
+      .pt-dark & {
+        color: map-get($pt-dark-intent-text-colors, $intent);
+      }
     }
   }
 }
 
-// styles for default .pt-tag-remove
-@mixin tag-remove() {
-  @include pt-icon();
-
-  position: absolute;
-  top: 0;
-  right: 0;
-  opacity: 0.5;
-
-  border: none;
-  background: none;
-  cursor: pointer;
-  padding: $tag-padding;
+@mixin custom-tag($color, $dark-color, $background-color) {
```

Comment:
I tried to do that yesterday, but it's actually not that easy because this mixin implement its own `pt-minimal` and `pt-dark` which would conflict with other usages above